src/main/asciidoc/_chapters/ops_mgt.adoc

```diff
@@ -2629,6 +2629,91 @@ You can use the HBase Shell command `status 'replication'` to monitor the replic
 * `status 'replication', 'source'` -- prints the status for each replication source, sorted by hostname.
 * `status 'replication', 'sink'` -- prints the status for each replication sink, sorted by hostname.
 
+==== Understanding the output
+
+The command output will vary according to the state of replication. For example right after a restart
+and if destination peer is not reachable, no replication source threads would be running,
+so no metrics would get displayed:
+
+----
+hbase01.home:
+SOURCE: PeerID=1
+Normal Queue: 1
+No Reader/Shipper threads runnning yet.
+SINK: TimeStampStarted=1591985197350, Waiting for OPs...
+----
+
+Under normal circumstances, a healthy, active-active replication deployment would
+show the following:
+
+----
+    hbase01.home:
+      SOURCE: PeerID=1
+         Normal Queue: 1
+           AgeOfLastShippedOp=0, TimeStampOfLastShippedOp=Fri Jun 12 18:49:23 BST 2020, SizeOfLogQueue=1, EditsReadFromLogQueue=1, OpsShippedToTarget=1, TimeStampOfNextToReplicate=Fri Jun 12 18:49:23 BST 2020, Replication Lag=0
+      SINK: TimeStampStarted=1591983663458, AgeOfLastAppliedOp=0, TimeStampsOfLastAppliedOp=Fri Jun 12 18:57:18 BST 2020
```

Comment:
> Just a question: For active-active replication, in order to get peerId of both clusters (defined in each other), we need to run status 'replication' at both clusters side right?

Yes. The command only shows the context of an individual cluster, listing overall stats about the given cluster source queues and sink threads.

>Getting ageOfLastShipped etc metric values from remote cluster is also not that easy even if we want to display here.

This "ageOfLastShipped" metric is related to the source cluster. On the source, we have _ReplicationSourceShipper_ thread reading entries from the WAL and making synchronous RPC calls to _ReplicationSink_ in the target. If the call is success, we get that time in _ReplicationSourceShipper_, decrease from the edit entry time and record it as "ageOfLastShipped". So "ageOfLastShipped" is how long a given edit took since it entered on source cluster until source cluster assumed it was successful replicated. 

@virajjasani would you think this metric description should be improved? Looks like current text was not that clear.

Reply:
Oh no, my bad. I know `ageOfLastShipped` metric, I just took it's example to state that we can't get similar metrics from destination cluster anyways.

My question was, do we really have any metric (from replication viewpoint) available to us from destination cluster? (I don't know of any such metric so far, I hope we can't know)

And by this command output, I was thinking what if we could display both clusters' metrics together (in case of active-active), but that might not be possible (and might not even be worth spending time)

Something like this could be really fancy but nothing necessary (hbase01.home belongs to cluster A and hbase02.home belongs to cluster B, only if we have 2 way replication setup):
```
    hbase01.home:
      SOURCE: PeerID=1
         Normal Queue: 1
           AgeOfLastShippedOp=0, TimeStampOfLastShippedOp=Fri Jun 12 18:49:23 BST 2020, SizeOfLogQueue=1, EditsReadFromLogQueue=1, OpsShippedToTarget=1, TimeStampOfNextToReplicate=Fri Jun 12 18:49:23 BST 2020, Replication Lag=0
      SINK: TimeStampStarted=1591983663458, AgeOfLastAppliedOp=0, TimeStampsOfLastAppliedOp=Fri Jun 12 18:57:18 BST 2020

    hbase02.home:
      SOURCE: PeerID=1
         Normal Queue: 1
           AgeOfLastShippedOp=0, TimeStampOfLastShippedOp=Fri Jun 12 18:49:23 BST 2020, SizeOfLogQueue=1, EditsReadFromLogQueue=1, OpsShippedToTarget=1, TimeStampOfNextToReplicate=Fri Jun 12 18:49:23 BST 2020, Replication Lag=0
      SINK: TimeStampStarted=1591983663458, AgeOfLastAppliedOp=0, TimeStampsOfLastAppliedOp=Fri Jun 12 18:57:18 BST 2020
```

Reply:
Anyways, even if one cluster can know another one's metrics, it's not related to this PR, it is good to merge anyways :)

Reply:
> My question was, do we really have any metric (from replication viewpoint) available to us from destination cluster? (I don't know of any such metric so far, I hope we can't know), like cluster A knows it's own ageOfLastShipped and ageOfLastApplied but does it know cluster B's ageOfLastShipped and ageOfLastApplied if both clusters are each other's pairs?

Ah, yeah, metrics from remote cluster are not available at source, and vice-versa. 

> And by this command output, I was thinking what if we could display both clusters' metrics together (in case of active-active), but that might not be possible (and might not even be worth spending time)

Maybe too much for shell command, but looks like a great idea for the replication stats page on the UI. Main problem I see is that it's not so easy to identify if a given cluster is actually a target, as it always exposes the sink rpc interface. I will dig further around and see if I can come with something not too complex.

Reply:
I agree, for shell, it is too much, and also the fact that determining whether the target cluster is also actively pushing WAL Edits to current cluster is not that straightforward.

Anyways, this was just a thought, maybe for some time in future, who knows we might have this status display feature :)